src/function/scalar/list/list_reverse.cpp

```diff
@@ -0,0 +1,124 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+
+namespace duckdb {
+
+static void ListReverseFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto count = args.size();
+
+	Vector &input_list = args.data[0];
+	if (input_list.GetType().id() == LogicalTypeId::SQLNULL) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::SetNull(result, true);
+		return;
+	}
+
+	// get the child vector
+	auto input_list_list_size = ListVector::GetListSize(input_list);
+	auto &input_list_child = ListVector::GetEntry(input_list);
```

Comment:
we're not interested in the list_child, so this can be left out

Reply:
We need input_list_child as we slice it later. Is there a way where we do not slice the child vector?

Reply:
By slicing the result child instead